lib/practice_exercises.rb

```diff
@@ -1,13 +1,63 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n), where n is the length of the string
+# The loop in this method get ran 1/2n
+# dropping the coefficient will be O(n)
+# Space Complexity: 0(1)
+# uses constant amount of memory
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # replace all non letters with empty string
+  new_string = string.upcase.gsub(/[^A-Za-z]/, '')
```

Comment:
Just a note, this `.upcase` and `.gsub` create new arrays leading to a space complexity of O(n).